metagenomics.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+import argparse
+import gzip
+import shutil
+import util.file
+import tools.kraken
+import tools.krona
+import tools.diamond
+__commands__ = []
+
+
+def kraken(inFastq1, inFastq2, db, outReport=None, outReads=None,
+           filterThreshold=None, threads=1):
+    assert outReads or outReport, (
+        'Either --outReads or --outReport must be specified.')
+
+    kraken_tool = tools.kraken.Kraken()
+    tmp_reads = util.file.mkstempfname('.kraken')
+    tmp_filtered_reads = util.file.mkstempfname('.filtered-kraken')
+    opts = {
+        '--paired': None,
+    }
+    # Could be optimized in 3.5 piping directly to kraken-filter.
+    kraken_tool.classify(db, [inFastq1, inFastq2], tmp_reads, options=opts)
+    opts = {
+        '--threshold': filterThreshold,
+    }
+
+    kraken_tool.execute('kraken-filter', db, tmp_filtered_reads, args=[tmp_reads],
+                        options=opts)
+
+    if outReads:
+        with open(tmp_filtered_reads, 'rb') as f_in:
+            with gzip.open(outReads, 'wb') as f_out:
+                shutil.copyfileobj(f_in, f_out)
+
```

Comment:
just to be safe, can we explicitly os.unlink(tmp_filtered_reads) at the end and any other large things? It shouldn't matter in a full invocation of the script as the tmpdir gets blown away at the end--unless something calls the function directly within python (e.g. nosetests)